api/applications.go

```diff
@@ -18,7 +18,7 @@ import (
 
 type ApplicationsClient interface {
 	GetApplication(ctx context.Context, clientID string) (Application, error)
-	CreateApplication(ctx context.Context, displayName string) (Application, error)
+	CreateApplication(ctx context.Context, displayName string, signInAudience string, tags []string) (Application, error)
```

Comment:
minor, perhaps worth a followup PR:
It might be worthwhile introducing a struct to hold the bag of parameters.
I typically use the convention of `*Input`, so in this case it would be:
```
type CreateApplicationInput struct {
...
```

Since this appears to be an exported interface, we don't want to break clients every time we add a new field, causing them to make updates to match the new signature.

Reply:
Sounds good, I will give this a shot in a followup